src/extract/faster_bottom_up.rs

```diff
@@ -0,0 +1,117 @@
+use super::*;
+
+/// A faster bottom up extractor inspired by the faster-greedy-dag extractor.
```

Comment:
Maybe explain if this is intended to give the same answers / implement an equivalent algorithm to bottom-up?

Reply:
I think it could help to describe this verbally too. My summary of faster-greedy-dag, and this algorithm, is something like:

> Bottom-up extraction works by iteratively computing the current best cost of each node in the e-graph based on the current best costs of its children. Extraction terminates when our estimates of the best cost for each node reach a fixed point. The baseline `bottom-up` implementation visits every node during each iteration of the fixed point. This algorithm instead _only_ visits the nodes whose current cost estimate may change: it does this by tracking parent-child relationships and storing relevant nodes in a work list (`UniqueQueue`).

Reply:
Thanks for that description! I put it in. Yes, it's supposed to give an equivalent algorithm